### PR TITLE
Added upload, migrated provider

### PR DESCRIPTION
**Pull request by Kayanski**
This PR aims at correcting the migration logic in case ibc client or ibc host versions are breaking

### Checklist

- [ ] CI is green.
- [ ] Changelog updated.

### CODE REVIEW

File: framework/packages/abstract-interface/src/migrate.rs [hunk 1]
```diff
@@ -246,7 +246,9 @@ impl<Chain: CwEnv> AbstractIbc<Chain> {
             &ibc_client_cw2_version,
             ::ibc_client::contract::CONTRACT_VERSION,
         ) {
-            // Version change is breaking, need to deploy new version
+            // Version change is breaking, need to upload and instantiate new version
+            self.client.upload_if_needed()?;
+            self.host.upload_if_needed()?;
             self.instantiate(&self.client.environment().sender_addr())?;
```
Comment (Buckram123) [inline framework/packages/abstract-interface/src/migrate.rs:252]: Oopsie!

File: interchain/scripts/state.json [hunk 2]
```diff
@@ -41,20 +41,20 @@
...
     },
     "default": {
       "abstract:account-local-0": "cosmos1hafge5xucuc03ar5lkwac0n9yctnxq6xrgygftdl9xl3tzncgghszwuw5z",
       "abstract:account-local-1": "cosmos1v9dl0fw5s3jujn6emmlgch7w4ynt02nykj75pnzk8mdw7nmu3p4q0fnq48",
       "abstract:ans-host": "cosmos1wx0qjtlz799pfxl73y2f4dv28nukztpmq2ztavk60v534tat9cdsvutp48",
-      "abstract:ibc-client": "cosmos1g86ckugm7v9vvpadq9szg6rlf72pdl4a6n0d8apltq5t0grecpwqg7qcv2",
-      "abstract:ibc-host": "cosmos1ldaf3e9zxja5mrcw3clcu873wggmqhwfvqw9djjcjxxjwv59crqsm8tma5",
+      "abstract:ibc-client": "cosmos1uspvkgrr9gfw429de02a5gx6dpl5neaw8srx4zsdd0xdtpk7gstq5ykc0u",
+      "abstract:ibc-host": "cosmos1x7q3st5m56g0h005l7eaas7ngh52tjsct2tmgs5hzcqlme0j604qwl604h",
       "abstract:module-factory": "cosmos1vnj802lyyupfaez73w7axfs3xveraxen370vhcnumaufj29r5rxqdm4ntk",
```
Comment (Buckram123) [inline interchain/scripts/state.json:58]: Do you know why new ibc got uploaded? Was it pre-release?

**Review by Buckram123 — APPROVED** [review]
Nice fix!

File: interchain/scripts/state.json (hunk 2 above)
Comment (Kayanski) [reply interchain/scripts/state.json:58]: Yes it was pre-release on provider

**Pull request merged**